File: src/domain/media.rs
use serde::{Deserialize, Serialize};
use time::OffsetDateTime;
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Media {
    pub id: Uuid,
    pub owner_id: Uuid,
    pub original_key: String,
    pub thumb_key: String,
    pub medium_key: String,
    pub width: i32,
    pub height: i32,
    pub bytes: i64,
    #[serde(with = "time::serde::rfc3339")]
    pub created_at: OffsetDateTime,
    pub thumb_url: Option<String>,
    pub medium_url: Option<String>,
    pub original_url: Option<String>,
}

File: spring/src/main/java/com/ciel/domain/Media.java
package com.ciel.domain;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;
import java.util.UUID;

/** Mirrors Rust's {@code domain::media::Media}. */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(JsonInclude.Include.ALWAYS)
public class Media {
    private UUID id;
    private UUID ownerId;
    private String originalKey;
    private String thumbKey;
    private String mediumKey;
    private int width;
    private int height;
    private long bytes;
    private OffsetDateTime createdAt;
    private String thumbUrl;
    private String mediumUrl;
    private String originalUrl;
}
